airbyte-commons-worker/src/main/java/io/airbyte/workers/helper/PersistConfigHelper.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.helper;
+
+import com.google.common.hash.Hashing;
+import io.airbyte.api.client.AirbyteApiClient;
+import io.airbyte.api.client.generated.DestinationApi;
+import io.airbyte.api.client.generated.JobsApi;
+import io.airbyte.api.client.generated.SourceApi;
+import io.airbyte.api.client.invoker.generated.ApiException;
+import io.airbyte.api.client.model.generated.ConnectionIdRequestBody;
+import io.airbyte.api.client.model.generated.ConnectionRead;
+import io.airbyte.api.client.model.generated.DestinationIdRequestBody;
+import io.airbyte.api.client.model.generated.DestinationRead;
+import io.airbyte.api.client.model.generated.DestinationUpdate;
+import io.airbyte.api.client.model.generated.JobIdRequestBody;
+import io.airbyte.api.client.model.generated.JobInfoLightRead;
+import io.airbyte.api.client.model.generated.SourceIdRequestBody;
+import io.airbyte.api.client.model.generated.SourceRead;
+import io.airbyte.api.client.model.generated.SourceUpdate;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.protocol.models.Config;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper class for workers to persist updates to Source/Destination configs emitted from
+ * AirbyteControlMessages.
+ */
+public class PersistConfigHelper {
```

Comment:
Extracted the methods to persist config updates for a job into this helper class since there are a few api calls here that we'll very likely re-use when bringing this functionality to the other connector commands.

Reply:
cc @davinchia think we should also have a `PersistStateHelper` and `PersistMetricsAndEstimates` helper like this?

Reply:
Yeap having a various persist helpers sounds generally right.

One thought - what about changing the name to `UpdateConnectorConfigHelper`? Update seems better as it's clear we are persisting to update and not persisting for the first time. We would propagate that name change to the functions here too e.g `updateSource` and `updateDestination`. Can probably drop the `config` suffix since `UpdateConnectorConfigHelper.updateSource` is pretty self-explanatory.

Can we also have a better explanation of why in the javadocs? This describes what and not really the why. We want to make it as easy as possible for a readers to piece the context together without talking to someone.

Reply:
I like it! Renamed and added some docs in [bff1d44](https://github.com/airbytehq/airbyte/pull/19811/commits/bff1d44900f3e682c37eb4adc6f9df6db77b2248)